src/main/kotlin/no/nav/syfo/service/microfrontend/MikrofrontendDialogmoteService.kt

```diff
@@ -0,0 +1,156 @@
+package no.nav.syfo.service.microfrontend
+
+import no.nav.syfo.db.*
+import no.nav.syfo.db.domain.toMikrofrontendSynlighet
+import no.nav.syfo.kafka.consumers.varselbus.domain.ArbeidstakerHendelse
+import no.nav.syfo.kafka.consumers.varselbus.domain.HendelseType
+import no.nav.syfo.kafka.consumers.varselbus.domain.getSynligTom
+import no.nav.syfo.kafka.producers.mineside_microfrontend.MikrofrontendSynlighet
+import no.nav.syfo.kafka.producers.mineside_microfrontend.MinSideRecord
+import no.nav.syfo.kafka.producers.mineside_microfrontend.Tjeneste
+import no.nav.syfo.service.microfrontend.MikrofrontendService.Companion.actionDisabled
+import no.nav.syfo.service.microfrontend.MikrofrontendService.Companion.actionEnabled
+import no.nav.syfo.utils.DuplicateMotebehovException
+import no.nav.syfo.utils.MotebehovAfterBookingException
+import no.nav.syfo.utils.VeilederAlreadyBookedMeetingException
+import org.slf4j.LoggerFactory
+
+class MikrofrontendDialogmoteService(
+    val database: DatabaseInterface
+) {
+    private val log = LoggerFactory.getLogger(MikrofrontendService::class.java)
+
+    companion object {
+        const val dialogmoteMikrofrontendId = "syfo-dialog"
+    }
+
+    fun updateDialogmoteFrontendForUserByHendelse(hendelse: ArbeidstakerHendelse, ferdigstill: Boolean): MinSideRecord? {
+        if (hendelse.type == HendelseType.SM_DIALOGMOTE_NYTT_TID_STED) {
+            setNewDateForMikrofrontendUser(hendelse)
+            return null
+        }
+
+        if (hendelse.type == HendelseType.SM_DIALOGMOTE_AVLYST || hendelse.type == HendelseType.SM_DIALOGMOTE_REFERAT) {
+            return minSideRecordDisabled(hendelse.arbeidstakerFnr)
+        }
+
+        val (userHasExistingDMEntries, userHasExistingMBEntries) =
+            userHasExistingMikrofrontendEntries(hendelse.arbeidstakerFnr)
+
+        throwExceptionOnIllegalDialogmoteUserState(
+            userHasExistingDMEntries,
+            userHasExistingMBEntries,
+            ferdigstill,
+            hendelse.type
+        )
+
+        if (hendelse.type == HendelseType.SM_DIALOGMOTE_INNKALT) {
+            if (userHasExistingMBEntries) {
```

Comment:
Kan vurdere å flytte logikken inne i dialogmøteinnkalling til egen funksjon, for å gjøre denne litt mindre

Reply:
Jepp 👍 Fikser